BrainPortal/app/models/boutiques_portal_task.rb

```diff
@@ -113,14 +118,19 @@ def before_form
         "#{iname} #{ioptional}\n"
       }.join("")
 
-    if num_in_files < num_needed_inputs || num_in_files > num_needed_inputs+num_opt_inputs
+    # Not enough files selected
+    if (num_in_files < num_needed_inputs)
       message = "This task requires #{num_needed_inputs} mandatory file(s) and #{num_opt_inputs} optional file(s)\n" +
         input_infos
       cb_error message
     end
 
-    #return "Warning: you selected more files than this task requires, so you won't be able to assign them all." if
-    #  num_needed_inputs > num_needed_inputs+num_opt_inputs
+    # Verify the number of files selected in case of a single mandatory file input or descriptor without a file input list
+    if ((!have_a_single_mandatory_input_file?(descriptor) && !have_file_input_list) && num_in_files > num_needed_inputs + num_opt_inputs)
```

Comment:
I don't understand, where does have_file_input_list come from? Is this a method, a variable? I don't know where it's defined?